internal/manifests/collector/podmonitor_test.go

```diff
@@ -43,17 +43,35 @@ func TestDesiredPodMonitors(t *testing.T) {
 	assert.Equal(t, fmt.Sprintf("%s-collector", params.OtelCol.Name), actual.Name)
 	assert.Equal(t, params.OtelCol.Namespace, actual.Namespace)
 	assert.Equal(t, "monitoring", actual.Spec.PodMetricsEndpoints[0].Port)
+	expectedSelectorLabels := map[string]string{
+		"app.kubernetes.io/component":  "opentelemetry-collector",
+		"app.kubernetes.io/instance":   fmt.Sprintf("%s.%s", params.OtelCol.Namespace, params.OtelCol.Name),
+		"app.kubernetes.io/managed-by": "opentelemetry-operator",
+		"app.kubernetes.io/part-of":    "opentelemetry",
+		"app.kubernetes.io/name":       fmt.Sprintf("%s-monitoring", params.OtelCol.Name),
+	}
+	assert.Equal(t, expectedSelectorLabels, actual.Spec.Selector.MatchLabels)
+}
 
-	params, err = newParams("", "testdata/prometheus-exporter.yaml")
+func TestDesiredPodMonitorsWithPrometheus(t *testing.T) {
+	params, err := newParams("", "testdata/prometheus-exporter.yaml")
 	assert.NoError(t, err)
 	params.OtelCol.Spec.Mode = v1alpha1.ModeSidecar
 	params.OtelCol.Spec.Observability.Metrics.EnableMetrics = true
-	actual, err = PodMonitor(params)
+	actual, err := PodMonitor(params)
 	assert.NoError(t, err)
 	assert.NotNil(t, actual)
 	assert.Equal(t, fmt.Sprintf("%s-collector", params.OtelCol.Name), actual.Name)
 	assert.Equal(t, params.OtelCol.Namespace, actual.Namespace)
 	assert.Equal(t, "monitoring", actual.Spec.PodMetricsEndpoints[0].Port)
 	assert.Equal(t, "prometheus-dev", actual.Spec.PodMetricsEndpoints[1].Port)
 	assert.Equal(t, "prometheus-prod", actual.Spec.PodMetricsEndpoints[2].Port)
+	expectedSelectorLabels := map[string]string{
+		"app.kubernetes.io/component":  "opentelemetry-collector",
+		"app.kubernetes.io/instance":   fmt.Sprintf("%s.%s", params.OtelCol.Namespace, params.OtelCol.Name),
+		"app.kubernetes.io/managed-by": "opentelemetry-operator",
+		"app.kubernetes.io/part-of":    "opentelemetry",
+		"app.kubernetes.io/name":       fmt.Sprintf("%s-monitoring", params.OtelCol.Name),
```

Comment:
we can't use the -monitoring suffix for the pod monitor, as that's the name for the service